spec/models/user_spec.rb

```diff
@@ -1,4 +1,15 @@
 require "rails_helper"
 
 RSpec.describe User, type: :model do
+  let(:user) { FactoryBot.build(:user) }
+
+  describe "validation tests" do
+    it "should save successfully" do
+      expect(user.save).to eq(true)
+    end
+  end
+
+  describe "association tests" do
+    it { is_expected.to have_one(:profile).with_dependent(:destroy) }
+  end
```

Comment:
We should also test if a profile was created if we use `after_create` callback

But if I remember well we decided that it is a great opportunity to learn ServiceObject, that will be responsible for creating a user with a profile